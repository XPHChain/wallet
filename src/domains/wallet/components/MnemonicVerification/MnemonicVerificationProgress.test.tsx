import React from "react";
import { render } from "utils/testing-library";

import { MnemonicVerificationProgress } from "./MnemonicVerificationProgress";

describe("MnemonicVerificationProgress", () => {
	it("should render tabs", () => {
		const positions = [1, 2, 3];
		const { getAllByTestId, asFragment } = render(
			<MnemonicVerificationProgress activeTab={1} wordPositions={positions} />,
		);
		const tabs = getAllByTestId("MnemonicVerificationProgress__Tab");

		expect(tabs).toHaveLength(positions.length);
		expect(asFragment()).toMatchSnapshot();
	});
});
