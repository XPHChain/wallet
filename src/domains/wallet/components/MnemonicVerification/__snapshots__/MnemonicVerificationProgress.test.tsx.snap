// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MnemonicVerificationProgress should render tabs 1`] = `
<DocumentFragment>
  <ul
    class="flex space-x-3"
  >
    <button
      class="MnemonicVerificationOptions__OptionButton-sc-qja4fq-0 MnemonicVerificationProgress__TabStyled-sc-m908j-0 fzvkGS bBMKbA"
      data-testid="MnemonicVerificationProgress__Tab"
      disabled=""
    >
      <span
        class="text-lg text-theme-success-600 dark:text-theme-secondary-200"
      >
        <div
          class="sc-bdnxRM NjstW"
          height="20"
          width="20"
        >
          <svg>
            circle-check-mark.svg
          </svg>
        </div>
      </span>
    </button>
    <button
      class="MnemonicVerificationOptions__OptionButton-sc-qja4fq-0 MnemonicVerificationProgress__TabStyled-sc-m908j-0 fzvkGS hhHogX"
      data-testid="MnemonicVerificationProgress__Tab"
      disabled=""
    >
      <span>
        The 2nd Word
      </span>
    </button>
    <button
      class="MnemonicVerificationOptions__OptionButton-sc-qja4fq-0 MnemonicVerificationProgress__TabStyled-sc-m908j-0 fzvkGS fnZKrA"
      data-testid="MnemonicVerificationProgress__Tab"
      disabled=""
    >
      <span>
        The 3rd Word
      </span>
    </button>
  </ul>
</DocumentFragment>
`;
